packages/auto-id/src/certificateManager.ts

```diff
@@ -0,0 +1,350 @@
+//! For key generation, management, `keyManagement.ts` file is used using "crypto" library.
+//! And for certificate related, used "@peculiar/x509" library.
+
+import {
+  blake2b_256,
+  concatenateUint8Arrays,
+  save,
+  stringToUint8Array,
+} from '@autonomys/auto-utils'
+import { AsnConvert } from '@peculiar/asn1-schema'
+import { AttributeTypeAndValue, GeneralNames } from '@peculiar/asn1-x509'
+import { Crypto } from '@peculiar/webcrypto'
+import * as x509 from '@peculiar/x509'
+import { KeyObject, createPublicKey } from 'crypto'
+import { doPublicKeysMatch, pemToPublicKey } from './keyManagement'
+import { randomSerialNumber } from './utils'
+
+const crypto = new Crypto()
+x509.cryptoProvider.set(crypto)
+
+interface SigningParams {
+  privateKey: CryptoKey
+  algorithm: 'sha256' | null // Only 'sha256' or null for Ed25519
+}
+
+export const OID_COMMON_NAME = '2.5.4.3' // OID for Common Name, not available in the library.
+const OID_SUBJECT_ALT_NAME = '2.5.29.17' // OID for Subject Alternative Name, not available in the library.
+
+export class CertificateManager {
+  private certificate: x509.X509Certificate | null
+  private privateKey: CryptoKey | null
+  private publicKey: CryptoKey | null
+
+  constructor(
+    certificate: x509.X509Certificate | null = null,
+    privateKey: CryptoKey | null = null,
+    publicKey: CryptoKey | null = null,
+  ) {
+    this.certificate = certificate
+    this.privateKey = privateKey
+    this.publicKey = publicKey
+  }
+
+  protected prepareSigningParams(): SigningParams {
+    const privateKey = this.privateKey
+
+    if (!privateKey) {
+      throw new Error('Private key is not set.')
+    }
+
+    if (privateKey.algorithm.name === 'Ed25519') {
+      return { privateKey: privateKey, algorithm: null }
+    }
+
+    if (privateKey.algorithm.name === 'rsa') {
+      return { privateKey: privateKey, algorithm: 'sha256' }
+    }
+
+    throw new Error('Unsupported key type for signing.')
+  }
+
+  protected static toCommonName(subjectName: string): x509.Name {
+    const commonNameAttr = new AttributeTypeAndValue({
+      type: OID_COMMON_NAME,
+      value: subjectName,
+    })
+    return new x509.Name([[commonNameAttr]])
+  }
+
+  // protected static toCommonName(subjectName: string): JsonName {
+  //   return [{ '2.5.4.3': [subjectName] }] // OID for commonName
+  // }
```

Comment:
Is there a reason to leave this code?

Reply:
Was using this as an alternative. Somehow missed removing it.